src/lib/getNetworkInformation.ts

```diff
@@ -0,0 +1,23 @@
+import { Navigator, NetworkInformation } from "../@types"
+
+/**
+ * Gets the Network Intoformation API interface via the navigator.connection
+ * global and clones the object as it is read only.
+ */
+export default function getNetworkInformation(): NetworkInformation {
+    const navWithInfo = navigator as Navigator
+    const connection = navWithInfo && navWithInfo.connection
+    if (connection) {
```

Comment:
We can use optional chaining here, which was introduced in TypeScript 3.7:
```suggestion
    const connection = navWithInfo?.connection
    if (connection) {
```